cw-orch-daemon/src/queriers/staking.rs

```diff
@@ -1,23 +1,33 @@
-use crate::{cosmos_modules, error::DaemonError};
+use crate::{cosmos_modules, error::DaemonError, Daemon};
 use cosmrs::proto::cosmos::base::query::v1beta1::PageRequest;
+use cw_orch_core::environment::{Querier, QuerierGetter};
 use tonic::transport::Channel;
 
-use super::DaemonQuerier;
-
 /// Querier for the Cosmos Staking module
+/// All the async function are prefixed with `_`
 pub struct Staking {
-    channel: Channel,
+    pub channel: Channel,
 }
 
-impl DaemonQuerier for Staking {
-    fn new(channel: Channel) -> Self {
+impl Staking {
+    pub fn new_async(channel: Channel) -> Self {
         Self { channel }
     }
 }
 
+impl Querier for Staking {
+    type Error = DaemonError;
+}
+
+impl QuerierGetter<Staking> for Daemon {
+    fn querier(&self) -> Staking {
+        Staking::new_async(self.channel())
```

Comment:
Can it not get the runtime handle from the daemon object?

Reply:
But it can. Keeping the same structure as for Queriers with traits